libtextsecure/outgoing_message.js

```diff
@@ -371,9 +393,35 @@ OutgoingMessage.prototype = {
       tempMessage.preKeyBundleMessage = this.message.preKeyBundleMessage;
       tempMessage.dataMessage = tempDataMessage;
       messageBuffer = tempMessage.toArrayBuffer();
+      logDetails = {
+        tempMessage,
+      };
     } else {
       messageBuffer = this.message.toArrayBuffer();
+      logDetails = {
+        message: this.message,
+      };
     }
+    const messageTypeStr = this.debugMessageType;
+
+    const ourPubKey = textsecure.storage.user.getNumber();
+    const ourPrimaryPubkey = window.storage.get('primaryDevicePubKey');
+    const secondaryPubKeys =
+      (await window.libloki.storage.getSecondaryDevicesFor(ourPubKey)) || [];
+    let aliasedPubkey = devicePubKey;
+    if (devicePubKey === ourPubKey) {
+      aliasedPubkey = 'OUR_PUBKEY'; // should not happen
+    } else if (devicePubKey === ourPrimaryPubkey) {
+      aliasedPubkey = 'OUR_PRIMARY_PUBKEY';
+    } else if (secondaryPubKeys.includes(devicePubKey)) {
+      aliasedPubkey = 'OUR SECONDARY PUBKEY';
+    }
+    libloki.api.debug.logSessionMessageSending(
+      `Sending ${messageTypeStr}:${
+        this.messageType
+      } message to ${aliasedPubkey} details:`,
+      logDetails
+    );
```

Comment:
log the type of message being sent with a redacted pubkey (SECONDARY or PRIMARY) if it is one of our device

Reply:
Uhm, is this for devs or are we redacting for the users when they submit to us?

Reply:
for dev, I found out it's way easier to debug rather than checking last 3 digits. when you start to have 2 linked devices syncing one with another and all being friend and in the same closed group, it makes exchanges way cleaner

Reply:
Maybe we should have a function for this logic and move it up the debug function call so it'll take in a string and substitute out our devices with string values.